### PR TITLE
Update param type

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
Update param type to Object instead of WPBlock.

Core ticket: https://core.trac.wordpress.org/ticket/53003

Props to ravipatel

### CODE REVIEW

**Comment by RaviPatel** [comment]
You probably have the wrong ravi mentioned here ;)

**Comment by mukeshpanchal27** [comment]
> You probably have the wrong ravi mentioned here ;)

I update the comment to remove mention.

File: packages/block-library/src/social-link/index.php [hunk 1]
```diff
@@ -8,9 +8,9 @@
 /**
  * Renders the `core/social-link` block on server.
  *
- * @param Array   $attributes The block attributes.
- * @param String  $content InnerBlocks content of the Block.
- * @param WPBlock $block Block object.
+ * @param Array  $attributes The block attributes.
+ * @param String $content InnerBlocks content of the Block.
+ * @param Object $block Block object.
```
Comment (mkaz) [inline packages/block-library/src/social-link/index.php]:
```suggestion
 * @param WP_Block $block Block object.
```

I believe this is a spill over from JS types, for PHP the type should be `WP_Block` and not a generic `Object` type.
You can see `WP_Block` in use for multiple other dynamic blocks [for example here](https://github.com/WordPress/gutenberg/blob/6156f72fd556a982c16423a7c8795ba997257518/packages/block-library/src/post-comments/index.php#L13).

**Review by mkaz — APPROVED** [review]
Thanks for updating:+1:

The PHP the type should be `WP_Block` and not a generic `Object` type. You can see `WP_Block` in use for multiple other dynamic blocks for [example here](https://github.com/WordPress/gutenberg/blob/6156f72fd556a982c16423a7c8795ba997257518/packages/block-library/src/post-comments/index.php#L13).

File: packages/block-library/src/social-link/index.php [hunk 2]
```diff
@@ -8,9 +8,9 @@
 /**
  * Renders the `core/social-link` block on server.
  *
- * @param Array   $attributes The block attributes.
- * @param String  $content InnerBlocks content of the Block.
- * @param WPBlock $block Block object.
+ * @param Array  $attributes The block attributes.
+ * @param String $content InnerBlocks content of the Block.
+ * @param WP_Block $block Block object.
```
Comment (mkaz) [inline packages/block-library/src/social-link/index.php]:
```suggestion
 * @param Array    $attributes The block attributes.
 * @param String   $content    InnerBlocks content of the Block.
 * @param WP_Block $block      Block object.
```

PHP doc rules 🙄

**Comment by mukeshpanchal27** [comment]
@mkaz thanks for the review and your thoughts.

**Pull request merged**